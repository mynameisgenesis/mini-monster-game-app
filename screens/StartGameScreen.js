import { useState } from "react";
import { TextInput, View, StyleSheet, Alert } from "react-native";
import Colors from "../utils/colors";
import PrimaryButton from "../components/ui/PrimaryButton";
import Title from "../components/ui/Title";
import Card from "../components/ui/Card";
import InstructionText from "../components/ui/InstructionText";

export default function StartGameScreen({ selectedNumberHandler }) {
  const [enteredNumber, setEnteredNumber] = useState("");

  function numberInputHandler(enteredText) {
    setEnteredNumber(enteredText);
  }

  function resetInputHandler() {
    setEnteredNumber("");
  }

  function confirmInputHandler() {
    const choosenNumber = parseInt(enteredNumber);
    if (isNaN(choosenNumber) || choosenNumber <= 0 || choosenNumber > 99) {
      Alert.alert(
        "üö® Invalid Number üö®",
        "Number needs to be between 1 and 99.",
        [{ text: "Okay üëç", style: "destructive", onPress: resetInputHandler }]
      );
      return;
    }

    selectedNumberHandler(choosenNumber);
  }

  return (
    <View>
      <Title>Guess My Number</Title>
      <Card>
        <InstructionText>Enter a Number</InstructionText>
        <TextInput
          testID="user-number-input"
          style={styles.numberInput}
          maxLength={2}
          keyboardType="number-pad"
          autoCapitalize="none"
          autoCorrect={false}
          onChangeText={numberInputHandler}
          value={enteredNumber}
        />
        {/* This view creates a new flexbox, that is why it is centered and the buttons resize */}
        <View style={styles.buttonsContainer}>
          <View style={styles.buttonContainer}>
            <PrimaryButton testID="reset-button" onPress={resetInputHandler}>
              Reset
            </PrimaryButton>
          </View>
          <View style={styles.buttonContainer}>
            <PrimaryButton
              testID="confirm-button"
              onPress={confirmInputHandler}
            >
              Confirm
            </PrimaryButton>
          </View>
        </View>
      </Card>
    </View>
  );
}

const styles = StyleSheet.create({
  numberInput: {
    height: 50,
    width: 50,
    textAlign: "center",
    fontSize: 32,
    borderBottomColor: Colors.secondary,
    borderBottomWidth: 2,
    marginVertical: 8,
    fontWeight: "bold",
    color: Colors.white,
  },
  buttonsContainer: {
    flexDirection: "row",
  },
  buttonContainer: {
    flex: 1,
  },
});
