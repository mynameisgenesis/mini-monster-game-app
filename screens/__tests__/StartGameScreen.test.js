import React from "react";
// import { create, act } from "react-test-renderer";
import StartGameScreen from "../StartGameScreen";
import {
  render,
  screen,
  fireEvent,
  cleanup,
} from "@testing-library/react-native";

jest.useFakeTimers();
jest.runAllTimers();

describe("StartGameScreen", () => {
  afterEach(cleanup);
  test("renders correctly", () => {
    expect(render(<StartGameScreen selectedNumberHandler={jest.fn()} />))
      .toMatchSnapshot;
//   });
//   test("input is cleared when reset button clicked ", () => {
//     const mockFunction = jest.fn();
//     render(<StartGameScreen selectedNumberHandler={mockFunction} />);
//     fireEvent(screen.findByTestId("user-number-input"), "onChangeText", "55");
//     expect(mockFunction).toHaveBeenCalledWith("55");
//   });
// });

// test("input is cleared when reset button pressed ", () => {
//   const userNumberInputField = tree.root.findByProps({
//     testID: "user-number-input",
//   });
//   expect(userNumberInputField).toBeDefined();
//   // userNumberInputField.value = 4;
//   const resetButton = tree.root.findByProps({ testID: "reset-button" }).props;
//   act(() => resetButton.onPress());
//   console.log(resetButton);
//   expect(userNumberInputField.value).toEqual("");
// });
