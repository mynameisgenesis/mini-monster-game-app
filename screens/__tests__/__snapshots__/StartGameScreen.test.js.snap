// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFD6EC",
      "borderRadius": 10,
      "elevation": 4,
      "justifyContent": "center",
      "marginHorizontal": 25,
      "marginTop": 100,
      "padding": 16,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 6,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    keyboardType="number-pad"
    maxLength={2}
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      {
        "borderBottomColor": "#FFA1CF",
        "borderBottomWidth": 2,
        "fontSize": 32,
        "fontWeight": "bold",
        "height": 50,
        "marginVertical": 8,
        "textAlign": "center",
        "width": 50,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "borderRadius": 30,
            "margin": 4,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#FF74B1",
              "elevation": 2,
              "paddingHorizontal": 8,
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "textAlign": "center",
              }
            }
          >
            Reset
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        style={
          {
            "borderRadius": 30,
            "margin": 4,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#FF74B1",
              "elevation": 2,
              "paddingHorizontal": 8,
              "paddingVertical": 8,
            }
          }
        >
          <Text
            style={
              {
                "color": "#FFFFFF",
                "textAlign": "center",
              }
            }
          >
            Confirm
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
